packages/@guardian/source-foundations/src/typography/data.ts

```diff
@@ -1,243 +1,153 @@
 import { pxToRem } from '../utils/px-to-rem';
-import type { Category, FontWeight, FontWeightDefinition } from './types';
 
-const fontSizes = [12, 14, 15, 17, 20, 24, 28, 34, 42, 50, 70] as const;
-
-const fonts = {
-	titlepiece: 'GT Guardian Titlepiece, Georgia, serif',
-	headlineSerif:
-		'GH Guardian Headline, Guardian Egyptian Web, Georgia, serif',
-	bodySerif:
-		'GuardianTextEgyptian, Guardian Text Egyptian Web, Georgia, serif',
-	bodySans:
-		'GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif',
-} as const;
-
-const lineHeights = [1.15, 1.35, 1.5] as const;
-
-const fontWeights = [300, 400, 500, 700] as const;
-
-const underlineThickness = [2, 3, 4, 5, 6] as const;
-
-const titlepieceSizes = {
-	small: fontSizes[8], //42px
-	medium: fontSizes[9], //50px
-	large: fontSizes[10], //70px
-} as const;
-
-const headlineSizes = {
-	xxxsmall: fontSizes[3], //17px
-	xxsmall: fontSizes[4], //20px
-	xsmall: fontSizes[5], //24px
-	small: fontSizes[6], //28px
-	medium: fontSizes[7], //34px
-	large: fontSizes[8], //42px
-	xlarge: fontSizes[9], //50px
-} as const;
-
-const bodySizes = {
-	xsmall: fontSizes[1], //14px
-	small: fontSizes[2], //15px
-	medium: fontSizes[3], //17px
-} as const;
-
-const textSansSizes = {
-	xxsmall: fontSizes[0], //12px
-	xsmall: fontSizes[1], //14px
-	small: fontSizes[2], //15px
-	medium: fontSizes[3], //17px
-	large: fontSizes[4], //20px
-	xlarge: fontSizes[5], //24px
-	xxlarge: fontSizes[6], //28px
-	xxxlarge: fontSizes[7], //34px
-} as const;
-
-const fontSizeMapping = {
-	titlepiece: titlepieceSizes,
-	headline: headlineSizes,
-	body: bodySizes,
-	textSans: textSansSizes,
-} as const;
-
-const remFontSizes = fontSizes.map((fontSize) => pxToRem(fontSize));
-
-const remTitlepieceSizes = {
-	small: remFontSizes[8], //42px
-	medium: remFontSizes[9], //50px
-	large: remFontSizes[10], //70px
-} as const;
-
-const remHeadlineSizes = {
-	xxxsmall: remFontSizes[3], //17px
-	xxsmall: remFontSizes[4], //20px
-	xsmall: remFontSizes[5], //24px
-	small: remFontSizes[6], //28px
-	medium: remFontSizes[7], //34px
-	large: remFontSizes[8], //42px
-	xlarge: remFontSizes[9], //50px
-} as const;
-
-const remBodySizes = {
-	xsmall: remFontSizes[1], //14px
-	small: remFontSizes[2], //15px
-	medium: remFontSizes[3], //17px
-} as const;
-
-const remTextSansSizes = {
-	xxsmall: remFontSizes[0], //12px
-	xsmall: remFontSizes[1], //14px
-	small: remFontSizes[2], //15px
-	medium: remFontSizes[3], //17px
-	large: remFontSizes[4], //20px
-	xlarge: remFontSizes[5], //24px
-	xxlarge: remFontSizes[6], //28px
-	xxxlarge: remFontSizes[7], //34px
+/**
+ * Pixel size values for each font that we use in the design system.
```

Comment:
how would feel about typing these things? something like:

```ts
type TextSize = 12 | 14 // etc
type FontSizeName = 'xxsmall' | 'xsmall' // etc

type PxTextSize = Record<FontSizeName, TextSize>
type PxTextSizes = Record<string, PxTextSize>

export const pxTextSizes: PxTextSizes = {
	textSans: {
		xxsmall: 12,
		xsmall: 14,
    }
}
```

Reply:
Completely agree — I had this in the back of my mind while doing this work. I added a note about this to the PR description too. My plan was to add as a followup PR so this one doesn't get too big :)

Reply:
i think it would be good to add them here, since they'll catch any accidental incorrect sizes etc in _this_ PR too

Reply:
👍  will include in this PR

Reply:
You will lose some information by dropping the `as const` assertion on `pxTextSizes`. If you type it as `PxTextSizes` you cannot then know that `xxxsmall` is not available on `headline`, for example.

One fix for this is to actually capture these in a test rather than via the compiler.

```ts
type sizes = [12, 14, 17,  /* … */];
type names = ["small", "medium", "large", /* … */ ];

test.each(Object.entries(pxTextSizes))("pxTextSizes has valid values for %s", (key, value) => {
	for (const [name, size] of Object.entries(value)) {
		expect(names).toContain(name);
		expect(sizes).toContain(size);
	}
})
```

Reply:
thanks for raising the PR - looks great!

Reply:
@mxdvl ah yeah that's a shame, i didn't realise that. (feels weird that you can't limit the values of a `const` assertion).

i wonder if that is more useful than pinning the pixels value down?

Reply:
That's all down to how the TypeScript compiler works. You cannot have it both ways, unfortunately. 

Either an object has a generic shape `Record<string, string>`, or it has a specific implementation `{ a: 0 } as const`. In this case I think knowing the exact mappings is valuable. It does mean you have to tell the compiler to chill whilst performing your tests.

Reply:
I've merged your PR @mxdvl  - thanks again for the contribution

Reply:
I'm a bit confused here by all the comments so apologies if this is irrelevant, but where we going to type some of these? I couldn't see the typings in the final diff?

Reply:
as I understood it, the resolution for this in the end was to check that the values are correct by making an assertion in the unit tests that Max put together in: https://github.com/guardian/source/pull/1535

Reply:
I've added to the comments on each (px and rem) object so people are aware of these tests: [`d727d94` (#1500)](https://github.com/guardian/source/pull/1500/commits/d727d9434739954e505b749a32592ce060caca20#diff-9f53feda13d0210df1793144b9787a8f820113e01c8b97eddaa9e391622d60eaR6)

Reply:
Nice one!